### PR TITLE
Closes issue #16945 Refactor TrackingProtectionOverlay to observe session via store.

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionOverlay.kt [hunk 1]
```diff
@@ -30,22 +42,47 @@ import org.mozilla.fenix.utils.Settings
...
+) : LifecycleAwareFeature {
+
+    @VisibleForTesting
+    internal var scope: CoroutineScope? = null
+
+    override fun start() {
+        store.flowScoped(lifecycleOwner) { flow ->
+            flow.mapNotNull { state ->
+                state.selectedTab
+            }.ifChanged { tab -> tab.content.loading }
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionOverlay.kt:62]: nit: formatting. \n :)

File: app/src/test/java/org/mozilla/fenix/trackingprotection/TrackingProtectionOverlayTest.kt [hunk 2]
```diff
@@ -30,74 +46,145 @@ class TrackingProtectionOverlayTest {
...
         every { toolbar.findViewById<View>(R.id.mozac_browser_toolbar_tracking_protection_indicator) } returns icon
     }
 
+    @After
+    fun cleanUp() {
+        testDispatcher.cleanupTestCoroutines()
+    }
+
+    @Test
+    fun `WHEN the loadingTHEN the overly is notified`() {
```
Comment (csadilek) [inline app/src/test/java/org/mozilla/fenix/trackingprotection/TrackingProtectionOverlayTest.kt:82]: nit: test name, maybe: WHEN loading state changes THEN overlay is notified....

**Review by csadilek — APPROVED** [review]
👍 🎉 this was the last observer :).

Just two nits below.

File: app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionOverlay.kt (hunk 1 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionOverlay.kt]: 👍🏽

File: app/src/test/java/org/mozilla/fenix/trackingprotection/TrackingProtectionOverlayTest.kt (hunk 2 above)
Comment (Amejia481) [reply app/src/test/java/org/mozilla/fenix/trackingprotection/TrackingProtectionOverlayTest.kt]: 👍🏽

**Comment by Amejia481** [comment]
Thanks for the feedback, I updated the pr with the changes :)

**Pull request merged**